tinyDA/chain.py

```diff
@@ -380,7 +417,14 @@ def _sample_coarse(self):
         # subsample the coarse model.
         for j in range(self.subchain_length):
             # draw a new proposal, given the previous parameters.
-            proposal = self.proposal.make_proposal(self.chain_coarse[-1])
+            if j==0:# use correct starting point for subchain
+                if self.accepted_fine[-1]:
+                    proposal = self.proposal.make_proposal(self.promoted_coarse[-1])
+                else: #if fine chain did not accept, jump back to start of last chain 
+                    proposal = self.proposal.make_proposal(self.chain_coarse[-(self.subchain_length+1)])
+
+            else:
```

Comment:
I don't think this is necessary, since the latest coarse link should already be the correct one, according to the acceptance check on the finer level.